samples/src/test/java/com/example/bigquery/ExtractTableToJsonIT.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.bigquery;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.cloud.storage.BucketInfo;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.testing.RemoteStorageHelper;
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+public class ExtractTableToJsonIT {
+  private ByteArrayOutputStream bout;
+  private PrintStream out;
+
+  @Before
+  public void setUp() throws Exception {
+    bout = new ByteArrayOutputStream();
+    out = new PrintStream(bout);
+    System.setOut(out);
+  }
+
+  @After
+  public void tearDown() {
+    System.setOut(null);
+  }
+
+  @Test
+  public void testExtractTableToJson() {
+    String projectId = "bigquery-public-data";
+    String datasetName = "samples";
+    String tableName = "shakespeare";
+    String bucketName = RemoteStorageHelper.generateBucketName();
+    String destinationUri = "gs://" + bucketName + "/extractTest.csv";
+
+    // Create GCS bucket to store extracted file
+    Storage storage = RemoteStorageHelper.create().getOptions().getService();
+    storage.create(BucketInfo.of(bucketName));
```

Comment:
This will lead to LOTS of different GCS buckets being created. Instead, consider using an environment variable to get the value for an existing bucket. You can use a function like [this](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/dlp/src/test/java/dlp/snippets/InspectTests.java#L39) to make it clear this resource is required to run the test.